### PR TITLE
fix(core): don't strip sourceMappingURL

### PR DESCRIPTION
**Pull request by asnowwolf**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

Strip all comment(include inline source mapping)

**What is the new behavior?**

Strip all comment but keep inline source mapping

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

related to #9664 

…om component styles

### CODE REVIEW

**Comment by vicb** [comment]
Don't strip sourceMappingURL would be a better commit comment IMO.

Thanks for your PR. Doesn't it fix an issue ?

**Comment by asnowwolf** [comment]
changed.

It fixed the #9664 .

File: modules/@angular/compiler/src/shadow_css.ts [hunk 1]
```diff
@@ -454,12 +455,20 @@ var _polyfillHostRe = new RegExp(_polyfillHost, 'im');
...
 
-var _commentRe = /\/\*[\s\S]*?\*\//g;
+var _commentRe = /\/\*\s*[\s\S]*?\*\//g;
 
 function stripComments(input:string):string {
   return StringWrapper.replaceAllMapped(input, _commentRe, (_: any /** TODO #9100 */) => '');
 }
 
+// all comments except inline source mapping ("/* #sourceMappingURL= ... */")
+var _sourceMappingUrlRe = /[\s\S]*(\/\*\s*#\s*sourceMappingURL=[\s\S]+?\*\/)\s*$/;
```
Comment (vicb) [inline modules/@angular/compiler/src/shadow_css.ts]: const

**Comment by vicb** [comment]
Thanks please change to const in new code (you can also update the current code if you feeel like- then do it in a refactor commit).

**Comment by vicb** [comment]
👍 thanks

**Comment by vicb** [comment]
Please amend the first commit to add fix #... to the message body

**Comment by asnowwolf** [comment]
there are many private functions could be refactor to standalone functions, should i refactor that after this?

**Comment by asnowwolf** [comment]
"Docs have been added / updated (for bug fixes / features)"

is there any doc about this issue? or just need to revise the inline docs?

**Comment by vicb** [comment]
I am just asking for you to update the **commit comment** so that the issue gets closed when we merge this

**Comment by asnowwolf** [comment]
o, I've changed to

	fix(core): Don't strip sourceMappingURL (#9664)

it's not right?

**Comment by vicb** [comment]
```
fix(core): Don't strip sourceMappingURL
fix #9664
```

> BTW: I can't understand the English very exactly.

neither can I :)

**Comment by asnowwolf** [comment]
In the commit guidelines, there should be a blank line between subject and body. should i follow it?

**Comment by asnowwolf** [comment]
done

**Comment by vicb** [comment]
many thanks !

> In the commit guidelines, there should be a blank line between subject and body. should i follow it?

If your commit message is plain text then yes you should. If you use any sort of GUI title and body should be separated anyway.

**Comment by asnowwolf** [comment]
My second PR will be merged :-) Thank you!

**Comment by vikerman** [comment]
Merged to master

**Pull request closed without merging**